tracing/trace_state.hh

```diff
@@ -179,6 +184,17 @@ public:
         return _state_props;
     }
 
+    /**
+     * @return the moment `begin()` was called, in microseconds from POSIX epoch.
+     */
+    uint64_t start_ts_us() const {
+        // `elapsed_clock` has undefined epoch, so we use the POSIX TS to expose times outside
+        const std::chrono::system_clock::time_point start_system_time_point = std::chrono::system_clock::now()
+                + _start.time_since_epoch()
+                - elapsed_clock::now().time_since_epoch();
```

Comment:
I'm a little disturbed by seeing now + time_since_epoch(). Parentheses around the subtraction would make it clear we're adding a delta to a time point, not a time point to a time point (which makes no physical sense).

In fact, with parentheses you can drop the time_since_epoch() calls, since the result of the subtraction is a duration, which can be added to time_point, as both physics and std::chrono allow.

Reply:
(it also takes care of scaling if needed)

Reply:
both done